src/main/proto/th2_grpc_common/common.proto

```diff
@@ -145,12 +153,32 @@ message ListValueFilter {
 }
 
 message MessageFilter {
-  string messageType = 1;
-  string direction = 2;
+  // Is not used for filtering sub-messages. Will be removed in the future
+  string messageType = 1 [deprecated = true];
+
+  // Is not used for filtering sub-messages. Will be removed in the future
+  string direction = 2 [deprecated = true];
   map<string, ValueFilter> fields = 3;
   ComparisonSettings comparison_settings = 4;
 }
 
+message MetadataFilter {
+  message SimpleFilter {
+    FilterOperation operation = 1;
+    bool key = 2;
+    string value = 3;
+  }
+
+  map<string, SimpleFilter> property_filters = 1;
+}
+
+message RootMessageFilter {
+  string messageType = 1;
+  map<string, ValueFilter> fields = 2;
```

Comment:
Maybe replace to MessageFilter message_filter

Reply:
It will simplify the migration but it also adds the extra nesting level...
Ok, I agree. Let's try to use `MessageFilter` instead of `map<string, ValueFilter>`